commands/operator-sdk/cmd/scorecard/scorecard.go

```diff
@@ -0,0 +1,252 @@
+		log.SetLevel(log.DebugLevel)
+	}
+	// if no namespaced manifest path is given, combine deploy/service_account.yaml, deploy/role.yaml, deploy/role_binding.yaml and deploy/operator.yaml
+	if SCConf.NamespacedMan == "" {
+		file, err := yamlutil.GenerateCombinedNamespacedManifest()
+		if err != nil {
+			log.Fatal(err)
+		}
+		SCConf.NamespacedMan = file.Name()
+		defer func() {
+			err := os.Remove(SCConf.NamespacedMan)
+			if err != nil {
+				log.Fatalf("could not delete temporary namespace manifest file: (%v)", err)
+			}
+		}()
+	}
+	if SCConf.GlobalMan == "" {
+		file, err := yamlutil.GenerateCombinedGlobalManifest()
+		if err != nil {
+			log.Fatal(err)
+		}
+		SCConf.GlobalMan = file.Name()
+		defer func() {
+			err := os.Remove(SCConf.GlobalMan)
+			if err != nil {
+				log.Fatalf("could not delete global manifest file: (%v)", err)
+			}
+		}()
+	}
+	defer cleanupScorecard()
+	var err error
+	kubeconfig, SCConf.Namespace, err = k8sInternal.GetKubeconfigAndNamespace(SCConf.KubeconfigPath)
+	if err != nil {
+		return err
+	}
+	scheme := runtime.NewScheme()
+	// scheme for client go
+	cgoscheme.AddToScheme(scheme)
+	// api extensions scheme (CRDs)
+	extscheme.AddToScheme(scheme)
+	// olm api (CSVs)
+	olmApi.AddToScheme(scheme)
+	dynamicDecoder = serializer.NewCodecFactory(scheme).UniversalDeserializer()
+	// if a user creates a new CRD, we need to be able to reset the rest mapper
+	// temporary kubeclient to get a cached discovery
+	kubeclient, err := kubernetes.NewForConfig(kubeconfig)
+	if err != nil {
+		return fmt.Errorf("failed to get a kubeclient: %v", err)
+	}
+	cachedDiscoveryClient := cached.NewMemCacheClient(kubeclient.Discovery())
+	restMapper = restmapper.NewDeferredDiscoveryRESTMapper(cachedDiscoveryClient)
+	restMapper.Reset()
+	runtimeClient, _ = client.New(kubeconfig, client.Options{Scheme: scheme, Mapper: restMapper})
+	err = createFromYAMLFile(SCConf.GlobalMan, false)
+	if err != nil {
+		return fmt.Errorf("failed to create global resources: %v", err)
+	}
+	err = createFromYAMLFile(SCConf.NamespacedMan, false)
+	if err != nil {
+		return fmt.Errorf("failed to create namespaced resources: %v", err)
+	}
+	err = createFromYAMLFile(SCConf.CrMan, true)
+	if err != nil {
+		return fmt.Errorf("failed to create cr resource: %v", err)
+	}
+	obj := unstructured.Unstructured{}
+	obj.SetAPIVersion(apiversion)
+	obj.SetKind(kind)
+	if SCConf.BasicTests {
+		fmt.Println("Checking for existence of spec and status blocks in CR")
+		err = checkSpecAndStat(runtimeClient, obj, false)
+		if err != nil {
+			return err
+		}
+		fmt.Println("Checking that operator actions are reflected in status")
+		err = checkStatusUpdate(runtimeClient, obj)
+		if err != nil {
+			return err
+		}
+		fmt.Println("Checking that writing into CRs has an effect")
+		logs, err := writingIntoCRsHasEffect(obj)
+		if err != nil {
+			return err
+		}
+		log.Debugf("Scorecard Proxy Logs: %v\n", logs)
+	} else {
+		err = checkSpecAndStat(runtimeClient, obj, true)
```

Comment:
Nit: a comment might be helpful here to explain why we're calling this even when basic tests are disabled.